### PR TITLE
add board/nucleo32-l031 adc configuration

### PR DESCRIPTION
**Pull request by paula75**
Add adc configuration for the nucleo32-l031, it has been tested on nucleo32-L031K6

### CODE REVIEW

File: boards/nucleo32-l031/include/periph_conf.h [hunk 1]
```diff
@@ -153,7 +153,15 @@ static const spi_conf_t spi_config[] = {
  * @name    ADC configuration
  * @{
  */
-#define ADC_NUMOF           (0)
+#define ADC_CONFIG {             \
+    { GPIO_PIN(PORT_A, 0), 0 },  \
```
Comment (aabadie) [inline boards/nucleo32-l031/include/periph_conf.h]: Please put comments about the on-board PIN label (`A0`, `A1`, etc). `A7` and `A5` are missing in the configuration btw.

**Comment by paula75** [comment]
ping

File: boards/nucleo32-l031/include/periph_conf.h [hunk 2]
```diff
@@ -153,7 +153,15 @@ static const spi_conf_t spi_config[] = {
  * @name    ADC configuration
  * @{
  */
-#define ADC_NUMOF           (0)
+#define ADC_CONFIG {             \
+    { GPIO_PIN(PORT_A, 0), 0 },  \ /* Pin A0 */
+    { GPIO_PIN(PORT_A, 1), 1 },  \ /* Pin A1 */
+    { GPIO_PIN(PORT_A, 3), 3 },  \ /* Pin A2 */
+    { GPIO_PIN(PORT_A, 4), 4 },  \ /* Pin A3 */
+    { GPIO_PIN(PORT_A, 7), 7 },  \ /* Pin A6 */
```
Comment (aabadie) [inline boards/nucleo32-l031/include/periph_conf.h]: This line should be at the end of the list to follow the pin labels on the board and [the mapping of arduino defines](https://github.com/RIOT-OS/RIOT/blob/master/boards/common/nucleo32/include/arduino_pinmap.h#L70) already available.
why not adding A5 and A7 ? They can be found in the board datasheet.

Comment (paula75) [reply boards/nucleo32-l031/include/periph_conf.h]: A7 is being used as tx in the uart. I added A5 to the list from now.

I'm not sure what you mean in the comment, is like this?
#define ADC_CONFIG {        \
    { ARDUINO_PIN_A0, 0 },  \
    { ARDUINO_PIN_A1, 1 },  \
    { ARDUINO_PIN_A2, 3 },  \
    { ARDUINO_PIN_A3, 4 },  \
    { ARDUINO_PIN_A4, 5 },  \
    { ARDUINO_PIN_A5, 6 },  \
    { ARDUINO_PIN_A6, 7 },  \
}

Comment (aabadie) [reply boards/nucleo32-l031/include/periph_conf.h]: ok, let's leave A7 out. Your change doesn't build with the comment outside the macro. Use this instead:
```c
{ GPIO_PIN(PORT_A, 1), 1 },  /* Pin A1 */ \
{ GPIO_PIN(PORT_A, 3), 3 },  /* Pin A2 */ \
{ GPIO_PIN(PORT_A, 4), 4 },  /* Pin A3 */ \
{ GPIO_PIN(PORT_A, 7), 7 },  /* Pin A6 */ \
```

Comment (paula75) [reply boards/nucleo32-l031/include/periph_conf.h]: fixed @aabadie

**Comment by aabadie** [comment]
let's go. Thanks @paula75 !

**Pull request merged**